Essentials/src/com/earth2me/essentials/commands/Commandmore.java

```
@@ -24,14 +24,34 @@ public void run(final Server server, final User user, final String commandLabel,
             throw new Exception(tl("fullStack"));
         }
         final String itemname = stack.getType().toString().toLowerCase(Locale.ENGLISH).replace("_", "");
-        if (ess.getSettings().permissionBasedItemSpawn() ? (!user.isAuthorized("essentials.itemspawn.item-all") && !user.isAuthorized("essentials.itemspawn.item-" + itemname) && !user.isAuthorized("essentials.itemspawn.item-" + stack.getTypeId())) : (!user.isAuthorized("essentials.itemspawn.exempt") && !user.canSpawnItem(stack.getTypeId()))) {
+        if (!user.canSpawnItem(stack.getType())) {
             throw new Exception(tl("cantSpawnItem", itemname));
         }
-        if (user.isAuthorized("essentials.oversizedstacks")) {
-            stack.setAmount(ess.getSettings().getOversizedStackSize());
+        if (args.length > 1) {
+            int newAmount = stack.getAmount();
+
+            try {
+                newAmount += Integer.parseInt(ChatColor.stripColor(args[1]));
+            } catch(Exception e) {
+                throw new Exception(tl("numberRequired"));
+            }
+
+            if (!user.isAuthorized("essentials.oversizedstacks")) {
+                newAmount = stack.getMaxStackSize();
+            } else {
+                if (newAmount > ess.getSettings().getOversizedStackSize()) {
```

Should we be capping this amount, or should we instead throw an error?

---

I think an error would be better here, as capping the amount might cause some confusion for users.

---

Nope, thats the correct way to do it. I'm also fairly certain you need to dig up translations for all of the translation variations.